rlclientlib/live_model_impl.cc

```diff
@@ -220,16 +221,25 @@ namespace reinforcement_learning {
     return error_code::success;
   }
 
-  int live_model_impl::request_multi_slot_decision(const char * context_json, unsigned int flags, multi_slot_response& resp, api_status* status)
+  int live_model_impl::request_multi_slot_decision(const char * context_json, unsigned int flags, multi_slot_response& resp, const std::vector<int>& baseline_actions, api_status* status)
   {
     const auto uuid = boost::uuids::to_string(boost::uuids::random_generator()());
-    return request_multi_slot_decision(uuid.c_str(), context_json, flags, resp, status);
+    return request_multi_slot_decision(uuid.c_str(), context_json, flags, resp, baseline_actions, status);
   }
 
-  int live_model_impl::request_multi_slot_decision(const char * event_id, const char * context_json, unsigned int flags, multi_slot_response& resp, api_status* status)
+  int live_model_impl::request_multi_slot_decision(const char * event_id, const char * context_json, unsigned int flags, multi_slot_response& resp, const std::vector<int>& baseline_actions, api_status* status)
   {
     resp.clear();
 
+    if (_learning_mode == APPRENTICE)
+    {
+      if (baseline_actions.size() == 0)
+      {
+        return error_code::baseline_actions_not_defined;
+      }
+      //TODO: validity check for baseline actions
```

Comment:
[nit] maybe this check can be moved between populate_multi_slot_response and log_decisions, where we have more informations about slots, but it can be done as followup

Reply:
I removed the comment for now. From what I remember we decided not to have a validity check for baselineactions on the RL client side (we have one on the APS side), but we can revisit this if needed.